src/quota.rs

```diff
@@ -0,0 +1,201 @@
+use anyhow::{anyhow, bail, Result};
+use async_imap::types::{Quota, QuotaResource};
+use humansize::{file_size_opts, FileSize};
+use itertools::Itertools;
+
+use crate::context::Context;
+use crate::imap::Imap;
+use crate::{
+    chat::{add_device_msg, add_device_msg_with_importance},
+    constants::Viewtype,
+    imap::scan_folders::get_watched_folders,
+    message::Message,
+};
+use crate::{
+    constants::DC_QUOTA_WARN_THRESHOLD_PERCENTAGE,
+    stock_str::{
+        quota_mailbox_nearly_full, quota_not_supported, quota_resource_messages,
+        quota_resource_storage,
+    },
+};
+
+/// Generates a detailed report about the current Quota usage on the for deltachat relevant folders
+/// and sends it to the user via [add_device_msg]
+///
+/// It's a bit like the prepaid mobile carrier service menu/messages,
+/// where you type a special number and then get a message back with your current balance.
+pub(crate) async fn quota_usage_report_job(context: &Context, imap: &mut Imap) -> Result<()> {
+    if !imap.capabilities_determined() {
```

Comment:
Just call `imap.prepare()` instead. It will ensure connection is established and capabilities are determined.

Reply:
actually this is just a guard to protect against using the function wrong, imap gets prepared before executing this function in the jobs function already (`generate_quota_usage_report`).

I'll change it so the imap preparation is done in this function instead of inside the job.